Classes/CRON/FormBuilder/Eel/Helper/TimeHashHelper.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace CRON\FormBuilder\Eel\Helper;
+
+use DateTime;
+use Neos\Flow\Annotations as Flow;
+use Neos\Eel\ProtectedContextAwareInterface;
+use Neos\Flow\Security\Cryptography\HashService;
+
+class TimeHashHelper implements ProtectedContextAwareInterface
+{
+
+    /**
+     * @Flow\Inject
+     * @var HashService
+     */
+    protected $hashService;
+
+    /**
+     * @return string a hashed timestamp to protect the form for spam bots
+     */
+    public function getTimeHash(): string
+    {
+        return $this->createTimeHash();
+    }
+
+    /**
+     * Creates an hashed timestamp
+     * @return string
+     */
+    protected function createTimeHash(): string
```

Comment:
Die Logik für das "Prüfen" des Timestamps gehört in eine separate Klasse. Die Funktionen `checkTimeStamp` und `createTimeHash` gehören dort rein und werden hier (und im Controller) nur noch aufgerufen. Sonst ist das Erzeugen und Prüfen im Code so voneinander entfernt, dass es schwer zu verstehen ist.

Evtl eine `class HoneyPotService` mit Funktionen von denen man nicht wissen muss, was sie "intern" machen: `getSecret` und `validateSecret` und dort wird die "Magic" mit dem Hash usw gemacht. Das macht es einfacher dieses mal zu ändern, ohne den Rest des Codes verstehen zu müssen.

Reply:
Hier gibt es jetzt eine separate Klassen, die das Übernimmt.